example/plugins/microservices/filter_attributes.yaml.example

```diff
@@ -2,6 +2,25 @@ module: satosa.micro_services.attribute_modifications.FilterAttributeValues
 name: AttributeFilter
 config:
     attribute_filters:
+        # default rules for any IdentityProvider
+        "":
+            # default rules for any requester
+            "":
+                # enforce controlled vocabulary (via simple notation)
+                eduPersonAffiliation: "^(faculty|student|staff|alum|member|affiliate|employee|library-walk-in)$"
+                eduPersonPrimaryAffiliation: "^(faculty|student|staff|alum|member|affiliate|employee|library-walk-in)$"
+                eduPersonScopedAffiliation:
+                    # enforce controlled vocabulary (via extended notation)
+                    regexp: "^(faculty|student|staff|alum|member|affiliate|employee|library-walk-in)@"
+                    # enforce correct scope
+                    shibmdscope_match_scope:
```

Comment:
These filters stand here without a key.
Wouldn't it be more natural to turn them into booleans?

The code would do:
```py
                elif filter_type == "shibmdscope_match_scope" and filter_value:
```

I know this is against the implementation logic (a loop over the filters) but I think from the point of view of a user it feels better.

Let me know.

Reply:
I was considering several options for how the filters would be specified.

I wanted to:
* allow selecting the filter type (going beyond just a regexp)
* allow using multiple filters for a single attribute (eduPersonScopedAffiliation needs both the md_scope filter and a regexp filter for the value).

I considered a list - like:
```
eduPersonScopedAffiliation:
  - type: regexp
    text: staff|student
  - type: shibmdscope_match_scope
```
... but thought that would be too verbose - and the list should not be required.

So went for a dict indexed by the filter type - and meaning of value is specific to the filter type.  And some filter types do not need a value at all.

So in the syntax I picked, it is actually a key with no value (from YAML parser, with value `None`).

I think I got influenced by Ansible when picking that - Ansible uses that a lot.  Ansible tasks get specified by including a dict key matching the name of the task - and if the task needs do parameters, you just have a key with no value - like here.

I could change that - but adding any value for these keys would just make the syntax more verbose, so having just the keys seems the most concise way to me.

Your thoughts?  (Or preferences on syntax?)

Reply:
OK, let's keep it as it is now.